src/util/fs.ts

```diff
@@ -11,3 +12,52 @@ export const isFileExists = async (filePath: string) => {
     return false;
   }
 };
+
+/**
+ * Checks if the directory exists with the given `filePath`.
+ */
+export const isDirectoryExists = async (directoryPath: string) => {
+  try {
+    await fs.access(directoryPath);
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Removes the given file from the directory.
+ */
+export const removeFileFromDirectory = async (filePath: string) => {
+  await fs.unlink(filePath);
+};
+
+/**
+ * Gets all files that have either .yml or .yaml extension in the given directory.
+ */
+export const getYamlFiles = async (directory: string) => {
+  let yamlFiles: string[] = [];
```

Comment:
have you tried like this/.

```
const files = await fs.readdir(directory);

  for (const file of files) {
    const fullPath = path.join(directory, file);
    const stat = await fs.lstat(fullPath);

    if (stat.isDirectory()) {
      return yamlFiles.concat(await getYamlFiles(fullPath));
    } 
    
    if (file.endsWith('.yml') || file.endsWith('.yaml')) {
      return yamlFiles.push(fullPath);
    }
  }

  return yamlFiles;
```

Reply:
@narekhovhannisyan , I've tried this way, but I've got some errors and left my version.